tests/server.py

```diff
@@ -593,7 +593,7 @@ def flush(self, maxbytes=None):
         if self.disconnected:
             return
 
-        if getattr(self.other, "transport") is None:
+        if not hasattr(self.other, "transport"):
```

Comment:
(copy from prev comment)

 From [this review chain](https://github.com/matrix-org/synapse/pull/9649#discussion_r597986898) I assume `hasattr` was originally intended here, instead of `getattr`.
 
 I accidentally changed this to just `hasattr` before, no `not`, which froze the tests.

Reply:
This seems like what this was trying to do, yes. 👍